### PR TITLE
WIP - do not merge - Adding logging to diagnose gold flake

### PR DESCRIPTION
**Pull request by Piinks**
## Description

Adds logging along the code path to adding a tryjob for Gold. This is currently flaking for multiple PRs. This should be reverted once the flake is fixed as it will intentionally add a lot of output to tests on Cirrus.

## Related Issues

Should help better identify what is going on in #49917

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by Piinks** [comment]
cc/ @fkorotkov @kjlubick @dnfield @christopherfujino Is there any other information I can include in the logging to help diagnose?

**Review by christopherfujino — APPROVED** [review]
lgtm

**Comment by kjlubick** [comment]
Not yet, but I hope to add a `goldctl whoami` that outputs who goldctl thinks it is and who the server thinks goldctl is to aid in debugging scenarios like this.

File: packages/flutter_goldens_client/lib/skia_client.dart [hunk 1]
```diff
@@ -611,6 +623,14 @@ class SkiaGoldClient {
     ));
     return authFile.existsSync();
   }
+
+  /// Returns the contents of the authorization configuration.
+  Future<String> _printClientAuthConfig() async {
+    return workDirectory.childFile(fs.path.join(
+      'temp',
+      'auth_opt.json',
```
Comment (dnfield) [inline packages/flutter_goldens_client/lib/skia_client.dart]: I'm assuming this doesnt' contain secrets? :)

Comment (Piinks) [reply packages/flutter_goldens_client/lib/skia_client.dart]: Nope. This is sample contents: 
```json
{
"Luci":false,
"ServiceAccount":"/dev/goldTesting/flutter-infra-56c2.json",
"GSUtil":false
}
```

I want to see what it is indicating for gsutil in particular.
But I think I will also add output for the `client_email` of the service account, which is what seems to be getting mixed up.

**Comment by Piinks** [comment]
I may not have to land this if I can trigger the flakes and get the necessary information.

**Comment by Piinks** [comment]
Ok, thankfully(?) it flaked.
It looks like this is happening when `auth_opt.json` (created by goldctl during auth) already exists. The SkiaClient checks to see if it is already there so it won't re-authenticate if unnecessary.

```dart
// in SkiaClient.auth
Skia Client is already authorized.
auth_opt.json: 
{
  "Luci":false,
  "ServiceAccount":"",
  "GSUtil":true
}
```

However, this does not happen every time, as there are other cases where everything seems to be ok:

```dart
// in SkiaClient.auth
Skia Client is already authorized.                                                                                                    
{
  "Luci":false,
  "ServiceAccount":"/tmp/flutter sdk/bin/cache/pkg/skia_goldens3735/packages/flutter/test/widgets/serviceAccount.json",
  "GSUtil":false
}
```

The `GOLD_SERVICE_ACCOUNT` seems to also be ok, confirmed the service account is present with the right information in every test (checked client_email).

**Comment by Piinks** [comment]
This is the log I've been referring to in comment above: https://api.cirrus-ci.com/v1/task/6064140468879360/logs/main.log

**Comment by Piinks** [comment]
Also posted this on the related issue:

I think I know what this is now. 🎉 

To solve #46687, #47551 added the empty auth option for `goldctl` so we could use the `check` feature. When using empty auth, no tryjobs are run/nothing is done that actually requires authentication.
I'm guessing that empty auth would create an auth_opt.json file like the first mentioned above where `GSUtil:true`

These files are all created in randomly generated directories to maintain thread safety.

If they are being cached, which I think they are (the 🐞 ), then future test runs that randomly choose the same directory that need to be authenticated don't because they see that the file is there and assume it's already set up, and therefore end up using gsutil.

I'll close this and put together a patch. :)

**Pull request closed without merging**